integrations/to.etc.domui.selenium/src/main/java/to/etc/domui/webdriver/core/base/DataTablePO.java

```diff
@@ -0,0 +1,91 @@
+package to.etc.domui.webdriver.core.base;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import to.etc.domui.webdriver.core.WebDriverConnector;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@NonNullByDefault
+public class DataTablePO extends BasePO {
+
+	private final String m_testId;
+
+	public DataTablePO(WebDriverConnector connector, String testId) {
+		super(connector);
+		m_testId = testId;
+	}
+
+	public int getNumberOfRows() {
+		return getRows().size();
+	}
+
+	private WebElement getTable() {
+		var table = wd().findElement(m_testId);
+		if(table == null) {
+			throw new IllegalStateException("cant find number list table");
+		}
+		return table;
+	}
+
+	public List<RowPO> getRows() {
+		return getTable().findElements(By.cssSelector("tbody tr")).stream().map(x->new RowPO(x, wd())).collect(Collectors.toUnmodifiableList());
+	}
+
+	public RowPO getRow(int i) {
+		return getRows().get(i);
+	}
+
+	public class RowPO {
+		private WebElement m_webElement;
+
+		private WebDriverConnector m_wd;
+
+		public RowPO(WebElement row, WebDriverConnector wd) {
+			m_webElement = row;
+			m_wd = wd;
+		}
+
+		public ColumnPO getColumn(int column) {
+			var columns =  getColumns();
+			return columns.get(column);
+		}
+
+		public List<ColumnPO> getColumns() {
+			return m_webElement.findElements(
+				By.cssSelector("td")
+			).stream().map(x->new ColumnPO(x,m_wd)).collect(Collectors.toUnmodifiableList());
+		}
+	}
+
+	public class ColumnPO {
+		private WebElement m_webElement;
+
+		private WebDriverConnector m_wd;
+
+		public ColumnPO(WebElement x, WebDriverConnector wd) {
+			m_webElement = x;
+			m_wd = wd;
+		}
+		public void click() {
+			m_webElement.click();
+		}
+		public void check() {
```

Comment:
Does not belong here and seems wrong, as this should be a "td", not a checkbox.

Reply:
I am actually not fond of this DataTable interface, because for me it's too low level. We should not do too much here until we have a better model in mind. My biggest gripe is with the index based access to columns and rows. That is error prone, I'd rather try something else.